### PR TITLE
plutonium core steal objective

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
added plutonium core steal objective for tators and everything it needs
if you have the objective you can get a susbox containing the gear for free in the new Objectives tab of the uplink

you can deconstruct, and reconstruct, nukes to and from a disarmed state

once you get the plutonium core you can put it into a nuke core container and close it, then it cannot be opened again (unless you emag it, the intention being if theres traitors and nukies at the same time its still somewhat possible for nukies to get the core out and rebuild the nuke. can also go the other way around of a traitor yoinking the nukies' core)

depends on #26779

## Why / Balance
less boring steal objectives that are just yoink/fishbomb someone for their gear, and more like mission impossible

## Technical details
- new `ObjectiveUnlockCondition` and `StoreUnlocker` pair added that let a listing only be accessible to someone that has an objective unlocking it
- added `SealableCabinet` that prevents opening the nuke core container when sealed

## Media

https://github.com/space-wizards/space-station-14/assets/39013340/0f007bd2-f133-4915-9426-a93e37874d77

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**
:cl:
- add: Traitors can now be assigned to deconstruct the station's nuke for its precious plutonium core.
- add: Added the Objectives category in the uplink with a Core Extraction Toolbox for deconstructing said nuke.

### CODE REVIEW

**Comment by HoidC** [comment]
demon core

**Comment by VektorZ1** [comment]
based

**Comment by deltanedas** [comment]
it is.

**Comment by deltanedas** [comment]
i didnt have a geiger counter which makes the sound, it does 4 rads/s

**Comment by deltanedas** [comment]
that was never a thing

**Comment by Emisse** [comment]
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

**Comment by deltanedas** [comment]
open

File: Content.Shared/Cabinet/SealingCabinetSystem.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        SubscribeLocalEvent<SealingCabinetComponent, OpenableOpenAttemptEvent>(OnOpenAttempt);
+        SubscribeLocalEvent<SealingCabinetComponent, GotEmaggedEvent>(OnEmagged);
+    }
+
+    private void OnOpenAttempt(Entity<SealingCabinetComponent> ent, ref OpenableOpenAttemptEvent args)
+    {
+        var cabinet = (ent, Comp<ItemCabinetComponent>(ent));
+        if (!_cabinet.HasItem(cabinet))
+            return;
+
```
Comment (metalgearsloth) [inline Content.Shared/Cabinet/SealingCabinetSystem.cs:27]: Comp will throw if they don't have the component so better to trycomp or omit entirely.

File: Content.Shared/Cabinet/SealingCabinetSystem.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    private void OnEmagged(Entity<SealingCabinetComponent> ent, ref GotEmaggedEvent args)
+    {
+        if (!ent.Comp.Emaggable)
+            return;
+
+        var cabinet = (ent, Comp<ItemCabinetComponent>(ent));
+        if (!_cabinet.HasItem(cabinet) || _openable.IsOpen(ent))
+            return;
+
```
Comment (metalgearsloth) [inline Content.Shared/Cabinet/SealingCabinetSystem.cs:41]: Same here.

**Review by metalgearsloth — APPROVED** [review]

File: Content.Shared/Cabinet/SealingCabinetSystem.cs (hunk 2 above)
Comment (deltanedas) [reply Content.Shared/Cabinet/SealingCabinetSystem.cs:41]: ,

File: Content.Shared/Cabinet/SealingCabinetSystem.cs (hunk 1 above)
Comment (Tayrtahn) [reply Content.Shared/Cabinet/SealingCabinetSystem.cs:27]: TryComp and logging an informative error on failure is probably preferable to throwing an exception.

Comment (metalgearsloth) [reply Content.Shared/Cabinet/SealingCabinetSystem.cs]: Any component can be added or removed at runtime, VV is indicative of this, and throwing should be avoided at all costs if it's expected.

**Comment by Moomoobeef** [comment]
there was literally an 11 hour window where this was merged. love it or hate it, that's pretty funny.

**Pull request merged**